src/ArchiveManagement/NexusMods.FileExtractor/IFileExtractorSettings.cs

```diff
@@ -21,26 +21,41 @@ public interface IFileExtractorSettings
 [PublicAPI]
 public class FileExtractorSettings : IFileExtractorSettings
 {
-    // Note: We can't serialize AbsolutePath because it contains more fields than expected. Just hope user sets correct paths and pray for the best.
-    private readonly IFileSystem _fileSystem;
-    
-    
+    /// <inheritdoc />
+    public ConfigurationPath TempFolderLocation { get; set; }
+
     /// <summary>
     /// Default constructor for serialization.
     /// </summary>
-    public FileExtractorSettings() : this(FileSystem.Shared) {}
+    public FileExtractorSettings() : this(FileSystem.Shared) { }
 
     /// <summary>
     /// Creates a default new instance of <see cref="FileExtractorSettings"/>.
     /// </summary>
     /// <param name="fileSystem"></param>
     public FileExtractorSettings(IFileSystem fileSystem)
     {
-        _fileSystem = fileSystem;
-        TempFolderLocation = new ConfigurationPath(_fileSystem
-            .GetKnownPath(KnownPath.EntryDirectory).Combine("Temp"));
+        TempFolderLocation = new ConfigurationPath(GetDefaultBaseDirectory(fileSystem));
+    }
+
+    private static AbsolutePath GetDefaultBaseDirectory(IFileSystem fs)
+    {
+        return fs.OS.MatchPlatform(
+            () => fs.GetKnownPath(KnownPath.LocalApplicationDataDirectory).Combine("NexusMods.App/Temp"),
+            () => fs.GetKnownPath(KnownPath.XDG_DATA_HOME).Combine("NexusMods.App/Temp"),
+            () => throw new NotSupportedException(
+                "(Note: Sewer) Paths needs PR for macOS. I don't have a non-painful way to access a Mac."));
+    }
```

Comment:
The temporary extraction folder should go somewhere else. `%Temp%` can be used on Windows without issues. On Linux things are a bit different due to `tmpfs` residing in memory and/or swap partition(s). However, I still think that `/tmp` should be used as a _default_ option, and we should probably have size limit checks in place anyways.

Reply:
I feel robbed 😭, you made this comment literally at the same time I made a commit to address this.

Reply:
In any case, are you sure `/tmp` is a good idea?

I had the same thing in mind, because `tmpfs` is in mem/swap, we risk a very easy case of memory starvation. 

Imagine for example the user queues downloads of multiple mods at once, with a large queue. `tmpfs` becomes 30+GB, and RAM/Swap are unable to handle all this space. Suddenly the system essentially grinds to a complete halt with the end user not knowing what really happened. 

But this happened directly after interacting with the App, thus we take the blame for it (not to mention more potential support requests). 

-------

Misc Note:

I've had a few cases of forced restarts earlier in App development, when tests didn't properly clean up after themselves. My `tmpfs` filled up without me noticing and only thing I had was the power button on my machine.

While the kernel does have an OOM killer and all, it is often pretty slow to react, and won't stop a user from halting the machine or by filling up that space by adding 'just another mod'.

Reply:
We discussed this over and decided to use XDG_STATE_HOME